python/python/lance/dataset.py

```diff
@@ -97,7 +97,9 @@ def execute(self, data_obj: ReaderLike, *, schema: Optional[pa.Schema] = None):
             source is some kind of generator.
         """
         reader = _coerce_reader(data_obj, schema)
-        super(MergeInsertBuilder, self).execute(reader)
+        inserted, updated, deleted = super(MergeInsertBuilder, self).execute(reader)
+
+        return inserted, updated, deleted
```

Comment:
Need to update the type signature of this function.

Also, could we maybe make this a dictionary? Like:

```python
{'num_rows_inserted': 10, 'num_rows_updated': 10, 'num_rows_deleted': 10}
```

Then we can add more metrics to the dictionary without breaking the API.